### DISCUSSION TITLE
How could this library work with Ember.js Octane?

### DISCUSSION THREAD

**Discussion by niamleeson** [LegendApp/legend-state#304] [Q&A]
I'm trying to understand how one could utilize this in the new version of EmberJs which has autotracking. The reason I'd like to use this library is because I'd like to extract the business logic out of the components. But in Ember, api calls are usually made with services. How would one inject services into observables, and how could one allow the component to re-render based on observable change?

**Comment by jmeistrich** [comment]
I’m not sure, I have no knowledge of Ember. Legend-State’s core should be adaptable to any framework, but it would need to be able to consume an external store. Does Ember have a way to consume data from external stores?

**Reply by niamleeson** [reply]
I believe ember-redux is using `@computed` decorator that tracks non-autotracked properties and re-runs getters like [this](https://ember-redux.com/octane/).
